### ISSUE TITLE
Adding support for Bokeh

### ISSUE THREAD

**Issue by bryevdv** [sphinx-gallery/sphinx-gallery#1009]
It would be great for `sphinx-gallery` to be able to handle Bokeh content. The Bokeh project docs currently use a custom gallery extension has been around since ~2012, which has very basic and limited functionality. An option to move to `sphinx-gallery` would be beneficial in several ways, not only to the Bokeh project, but also to general users who have asked for this capability in the past. 

As I understand things, the main task would to create a suitable custom scraper that can embed the Bokeh Javascript content into Sphinx output, and also pull pre-supplied thumbnails (by default) or run PNG export functions to generate thumbnails (optionally). 

The Bokeh project does have some project funds currently available, so it is potentially possible that this work could be supported financially (through a NumFocus contractor agreement).

**Comment by lucyleeow** [comment]
>  suitable custom scraper that can embed the Bokeh Javascript content into Sphinx output,

Would this: https://sphinx-gallery.github.io/stable/configuration.html#capture-repr be helpful at all?

> The Bokeh project does have some project funds currently available, so it is potentially possible that this work could be supported financially (through a NumFocus contractor agreement).

Do you have someone in mind or are you looking for suggestions?

**Comment by bryevdv** [comment]
> Would this: https://sphinx-gallery.github.io/stable/configuration.html#capture-repr be helpful at all?

I don't think so, the current `_html_repr` provides an expandable textual view of the model and its properties, not an embeddable plot (or whatever), and that is unlikely to change. We prefer to have an explicit `show(plot)` in notebooks. The current `:bokeh-plot:` sphinx extensions mokey-patches `show` to grab the thing to embed, and generates a self-contained sidecar JS script that embeds the plot. 

> Do you have someone in mind or are you looking for suggestions?

Looking for suggestions, e.g. anyone who is currently active with sphinx-gallery dev (or otherwise with relevant experience) who might be interested. We can certainly provide technical expertise on the Bokeh side of things.

**Comment by HealthyPear** [comment]
Hello, I just stumbled upon this missing support and I have managed to do something basic, but I have encountered some of the limitations that you describe (the thumbnail for example).

I am using sphinx gallery for one of my projects and I managed to use the bokeh_plot directive from the docstring of the example file. Of course the thumbnail is not catched and I get the default image from SG and the example counts actually as failed for some reason because there is no content outside of the docstring.

Anything else one could do more than this while any specific support is designed?
